File: Solution/src/com/solution/dynamicProgramming/LongestIncreasingSubset.java
package com.solution.dynamicProgramming;

import java.util.Scanner;

import com.solution.main.ReadInput;

public class LongestIncreasingSubset {
	public void run() {

		Scanner sc = ReadInput.getScanner("longestIncreasingSubset");

		int size = sc.nextInt();

		int[] arr = ReadInput.getIntArray(size, sc.next());

		System.out.println(doRun(arr));
	}

	private int doRun(int[] arr) {

		int[] T = new int[arr.length];

		for (int i = 0; i < T.length; i++) {
			T[i] = 1;
		}

		for (int i = 1; i < arr.length;i++) {
			for (int j = 0; j < i; j++) {
				if (arr[j] < arr[i]) {
					T[i] = Math.max(T[i], T[j] + 1);
				}
			}
		}

		int max = T[0];
		for(int i = 1 ; i < T.length; i ++){
			max = Math.max(max,T[i]);
		}
		
		return max;
	}
	
	
	

}

File: src/main/kotlin/com/cssolutions/dynamicProgramming/LongestIncreasingSubset.kt
package com.cssolutions.dynamicProgramming

import com.cssolutions.main.ReadInput

class LongestIncreasingSubset {
    fun run() {
        val sc = ReadInput.getScanner("longestIncreasingSubset")
        val size = sc.nextInt()
        val arr = ReadInput.getIntArray(size, sc.next())
        println(doRun(arr))
    }

    private fun doRun(nums: IntArray): Int {
        val numsLIS = IntArray(nums.size) { 1 }
        for (i in nums.indices) {
            for (j in i - 1 until 0 step -1) {
                if (nums[j] < nums[i])
                    numsLIS[i] = Math.max(numsLIS[i], nums[j] + 1)
            }
        }

        return numsLIS.maxOrNull() ?: 1
    }
}